### PR TITLE
feat(serviceinfo,tes): add models

### PR DESCRIPTION
**Pull request by aaravm**
This PR contains all the autogenerated models created by running the script in the #27 
This PR doesn't contain any manually written code, and everything is autogenerated

## Summary by Sourcery

Add autogenerated models for the Task Execution Service (TES) API and GA4GH service-info API. Introduce scripts to automate the generation of these models from OpenAPI specifications.

New Features:
- Add autogenerated models for the Task Execution Service (TES) API, including TesServiceInfo, TesTask, TesExecutor, TesResources, TesInput, TesOutput, TesState, TesTaskLog, TesExecutorLog, TesOutputFileLog, and TesCreateTaskResponse.
- Add autogenerated models for the GA4GH service-info API, including Service, ServiceOrganization, and ServiceType.

Enhancements:
- Add a script (utils/script.sh) to automate the generation of models from OpenAPI specifications using the OpenAPI Generator CLI.
- Add a build script (utils/build-models.sh) to generate models for multiple APIs by calling the main script with appropriate parameters.

### CODE REVIEW

File: lib/src/tes/models/tes_state.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+pub enum TesState {
```
[the remark this answers is not in the document]
Comment (aaravm) [reply lib/src/tes/models/tes_state.rs]: Not our concern, will be handled by the OpenAPI specs

File: utils/script.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+# Exit immediately if a command exits with a non-zero status.
+set -euo pipefail
+
+# Function to ensure the OpenAPI Generator JAR file is set up
+ensure_openapi_generator() {
+    mkdir -p ~/bin/openapitools
+    if [ ! -f "$OPENAPI_GENERATOR_JAR" ]; then
+        curl -L "$OPENAPI_GENERATOR_URL" -o "$OPENAPI_GENERATOR_JAR"
+        echo "d41d8cd98f00b204e9800998ecf8427e  $OPENAPI_GENERATOR_JAR" | sha256sum -c -
```
[the remark this answers is not in the document]
Comment (aaravm) [reply utils/script.sh]: File not in PR

**Review by uniqueg — CHANGES_REQUESTED** [review]
I have only one complaint here: We currently dump a lot of stuff in the `src/` package root directory. Across the various PRs there is `lib.rs`, `test_utils.rs`, `configuration.rs`, transport.rs`. Now we add a folder for each supported client/spec - which may end up being many (supported out of the box, plus whatever others like to add when they extend the SDK).

I would suggest to add some sub-packages: Certainly one for `clients/` which holds `tes/`, `/serviceinfo/` etc., which you could introduce here.

And maybe also one for `utils/` (`transport.rs` could go in there, and also `test_utils.rs`, perhaps renamed to `testing.rs` - given that from the package name it becomes clear that it is a utility module). Of course, changes to these should go to other PRs (e.g., #30 etc., or a whole new PR).

It may not sound super important now, but changing the package structure later on is a lot more of a hassle than it is now.

**Comment by aaravm** [comment]
Ok, I will move `transport` and `configuration` to `utils` folder, and will move `tes` and `serviceinfo` to a `clients` folder in further PRs. 
@uniqueg, I have merged #30, and have moved to a `utils` folder in this PR, and moved the models to `clients`, since the comment regarding changing folders is mentioned here. I think this PR is ready to be merged.

**Review by uniqueg — APPROVED** [review]

**Pull request merged**